src/deploy/functions/discovery/jsexports/parseTriggers.ts

```diff
@@ -227,7 +227,9 @@ export function addResourcesToBackend(
         cloudFunction.trigger.eventFilters.resource = `${cloudFunction.trigger.eventFilters.resource}/${id}`;
       }
 
-      cloudFunction.labels = { "deployment-scheduled": "true" };
+      cloudFunction.labels = Object.assign(cloudFunction.labels, {
```

Comment:
Nit: now that we have object destructuring I personally find it cleaner to use that syntax:

```js
cloudfunctions.labels = {
   ...cloudfunctions.lables,
   "deployment-scheduled": true,
}
```
   
Or at least I would if we wanted to create a copy of the dictionary. In this case, your code is equivalent to:

```js
cloudFunction.labels["deployment-scheduled"] = true;
```

Reply:
I prefer that too, so I'll switch this over. As is, this is not quite equivalent to cloudFunction.labels["deployment-scheduled"], because labels is optional & can be undefined. Object.assign safely handles that case, wheres the above does not.

Reply:
```
> obj = {}
{}
> obj.labels = Object.assign(obj.lables, {"foo": "bar"});
Uncaught TypeError: Cannot convert undefined or null to object
    at Function.assign (<anonymous>)
```

Object splat supports undefined but `Object.assign` does not.